### PR TITLE
Move entries that stick out too far on the wrong side to the front

### PR DESCRIPTION
**Pull request by alloy**
This is a fix for https://github.com/artsy/eigen/issues/301

Currently the heuristic I chose is that an item may not stick out more than 50% _of its own height_ compared to the shortest section _in front of it_.

There are a few inset related differences when integrating into Eigen that I still need to fix. I think this means there are some tests missing in this repo around the insets, so I’ll get that done first.

### CODE REVIEW

File: ARCollectionViewMasonryLayout.podspec [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
-  s.author           = { "Orta Therox" => "orta.therox@gmail.com" }
-  s.social_media_url = "http://twitter.com/orta"
-  s.platform         = :ios
-  s.platform         = :ios, '7.0'
-  s.source           = { :git => "https://github.com/AshFurrow/ARCollectionViewMasonryLayout.git", :tag => "#{s.version}" }
-  s.source_files     = 'ARCollectionViewMasonryLayout.{h,m}'
-  s.framework        = 'UIKit'
-  s.requires_arc     = true
+  s.name                = "ARCollectionViewMasonryLayout"
+  s.version             = "2.0.0"
```
Comment (orta) [inline ARCollectionViewMasonryLayout.podspec]: is this API compatible?

Comment (alloy) [inline ARCollectionViewMasonryLayout.podspec]: It sure is API-wise (see the [changes to the public header](https://github.com/ashfurrow/ARCollectionViewMasonryLayout/pull/23/files#diff-17e892cfe7f27304974edd320ccddc46)), but the behaviour does change of course, so not sure how we should be labelling that under SemVer.

Comment (orta) [inline ARCollectionViewMasonryLayout.podspec]: IMO, lets 2.1.0 it, seems like a good feature for me

**Comment by alloy** [comment]
Btw, [this previously existing snapshot](https://github.com/ashfurrow/ARCollectionViewMasonryLayout/pull/23/files#diff-6151ae3fc1da49b396b7cba51672423a) is a good example of how the behaviour is changed.

**Comment by alloy** [comment]
This should be good to merge now :+1:

**Comment by ashfurrow** [comment]
Cool – I'll take a final look-through.

**Comment by ashfurrow** [comment]
I've seen similar errors to the one on Travis when I update pod version numbers. I don't know if this is the best fix, but I just remove the development pod from the podfile, pod install, add it back in, and then pod install again.

**Comment by orta** [comment]
Yeah  it needs a local `pod update` ( or to always call that in the CI? )

File: ARCollectionViewMasonryLayout.podspec [hunk 2]
```diff
@@ -1,16 +1,17 @@
...
+  s.version             = "2.1.0"
+  s.summary             = "ARCollectionViewMasonryLayout is a UICollectionViewLayout subclass for creating flow-like layouts with dynamic widths or heights."
+  s.homepage            = "https://github.com/AshFurrow/ARCollectionViewMasonryLayout"
+  s.screenshots         = "https://raw.githubusercontent.com/AshFurrow/ARCollectionViewMasonryLayout/master/Screenshots/ARCollectionViewMasonryLayout.png"
+  s.license             = 'MIT'
+  s.author              = { "Orta Therox" => "orta.therox@gmail.com" }
+  s.social_media_url    = "http://twitter.com/orta"
+  s.platform            = :ios
+  s.platform            = :ios, '7.0'
+  s.source              = { :git => "https://github.com/AshFurrow/ARCollectionViewMasonryLayout.git", :tag => "#{s.version}" }
```
Comment (ashfurrow) [inline ARCollectionViewMasonryLayout.podspec]: Can probably take out `"#{s.version}"` (totally my fault).

**Comment by ashfurrow** [comment]
Looks great. Big fan of the abstraction you used. Going to merge, then fix the Podfile on master.

**Comment by ashfurrow** [comment]
2.1.0 is pushed to trunk.

**Comment by alloy** [comment]
Ah yeah, that makes sense. I probably shouldn’t have bumped the version for you yet, sorries.

Glad you like it :) There was also room for an abstraction around the various metrics in horizontal vs vertical mode, but didn’t yet spent time on that. E.g. a `Metric` class that’s initialised with the direction and the various sizes/insets and then returns the right ones. That instance can then be shared between the layout and the grid.

**Pull request merged**